### ISSUE TITLE
can things go in other places?

### ISSUE THREAD

**Issue by shmish111** [Heather/Control.Eternal.Idris#2]
I have been looking at things in https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/idris-modules that are not working fully and can be cleaned up. Eternal doesn't seem to build with nix on mac, probably due to some of the C stuff. So I had a look at eternal to see if I could fix it and saw that it's basically a collection of utility functions so my question is, could these functions go in more suitable libraries or maybe even in Prelude? Also, what does `readProcess` do and what sort of library would it fit into? I think some functions may not be needed anymore, for example `<@>` is just `<$>` and `for` is just `traverse` I think.

**Comment by cnd** [comment]
@shmish111 hello, I didn't even knew that it was packaged there. yes it's collection of various utilities which I've used in few minor toy projects. `readProcess` is used here https://github.com/Heather/Control.Eternal.Idris/blob/master/Control/Eternal/System/Process.idr - it simply runs an command and reads out it's output into `IO String`.
Yes it's possible to avoid many helper functions from that library, for is really just traverse, many things were added when I were learning and not changed, I don't think that anything from this library will be welcomed anywhere else, split is possible but not sure still.
( I wonder why that one is not packaged https://github.com/Heather/idris-unicode >_< )

**Comment by shmish111** [comment]
I think someone at some time probably searched github for any Idris projects and added them, possibly not very nice given the author becomes a maintainer without asking :-O

Anyway, would you like me to get Eternal removed from nixpkgs and try to build and add idris-unicode?

If you ever find yourself using Eternal again could you see if you can find a maintained-ish library that whatever you need would fit into and raise a PR for that?

**Comment by cnd** [comment]
@shmish111 eternal is used for my two afair other toy projects however I've never used nix... feel free to add unicode =) and for current library I don't care much, it will be better to ask Silvan or Márton =)

**Comment by shmish111** [comment]
What is Heather, who are Silvan and Márton? Also, do you use windows? what about WSL, nix apparently works on WSL and I was looking for someone to test idris on nix on windows.

**Comment by cnd** [comment]
@shmish111 and WSL is extra slow (specially when working with fs), look the best Idris package manager: https://github.com/Heather/Synthia/blob/master/Synthia.idr (okay, I'm joking) there actually were something better than that which are listed there https://github.com/idris-lang/Idris-dev/wiki/Build-tools-%7C-Package-managers and also

**Comment by shmish111** [comment]
Yeah, package managers are big and complex if you want them to be useable and nix and bazel have already solved loads of problems and are well maintained, that's why I think it would be better to use these rather than make yet another package manager.
